extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/DevServicesLambdaProcessor.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.amazon.lambda.deployment;
+
+import static io.quarkus.deployment.annotations.ExecutionTime.STATIC_INIT;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.amazon.lambda.runtime.AmazonLambdaApi;
+import io.quarkus.amazon.lambda.runtime.LambdaHotReplacementRecorder;
+import io.quarkus.amazon.lambda.runtime.MockEventServer;
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.DevServicesNativeConfigResultBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.ServiceStartBuildItem;
+import io.quarkus.runtime.LaunchMode;
+
+public class DevServicesLambdaProcessor {
+    private static final Logger log = Logger.getLogger(DevServicesLambdaProcessor.class);
+
+    static MockEventServer server;
+    static boolean first = true;
+
+    @BuildStep(onlyIfNot = IsNormal.class)
+    @Record(STATIC_INIT)
+    public void enableHotReplacementChecker(LaunchModeBuildItem launchMode,
+            LambdaHotReplacementRecorder recorder,
+            LambdaObjectMapperInitializedBuildItem dependency) {
+        if (launchMode.getLaunchMode().isDevOrTest()) {
+            try {
+                // if deprecated test framework in classpath, do not enable
+                Class.forName("io.quarkus.amazon.lambda.test.LambdaClient");
+            } catch (ClassNotFoundException e) {
+                recorder.enable();
+            }
+        }
+    }
+
+    @BuildStep(onlyIfNot = IsNormal.class)
+    public void startEventServer(LaunchModeBuildItem launchMode,
+            Optional<EventServerOverrideBuildItem> override,
+            BuildProducer<DevServicesNativeConfigResultBuildItem> devServicePropertiesProducer,
+            BuildProducer<ServiceStartBuildItem> serviceStartBuildItemBuildProducer) throws Exception {
+        if (!launchMode.getLaunchMode().isDevOrTest())
+            return;
+
+        if (server != null) {
+            if (launchMode.getLaunchMode() == LaunchMode.TEST) {
+                server.close();
+                server = null;
+            } else {
+                return;
+            }
+        }
+        Supplier<MockEventServer> supplier = null;
+        if (override.isPresent()) {
+            supplier = override.get().getServer();
+        } else {
+            supplier = () -> new MockEventServer();
```

Comment:
You should probably have a proper config object for the port here and pass it into the mock event server. You also need a different port for testing/dev to allow continuous testing to work.

Reply:
@stuartwdouglas A system property sets the port.  Why does it have to be a different port for continuous testing to work?  Does continuous testing have its only HTTP server?

Reply:
When you run `mvn quarkus:dev` and then enable continuous testing you will end up with two MockServer instances, one for dev mode and one for the tests. For HTTP dev mode is 8080 and tests is 8081.

`mvn quarkus:test` would still work with a single test, but not the usual dev mode experience.

Reply:
@stuartwdouglas Really confused on what you mean here that there will be two MockServers.  When I do quarkus:dev, I see it launch my MockServer on 8081.  I'm able to hit it with curl.  I'm able to hit 'r' and start test mode.  I'm still able to curl on 8081.  Why would I want to run two mock servers?  I don't think something like Kafka is doing this either and I"ve been modeling my code after that DevService

Reply:
Kafka and the databases start multiple instances. Basically this keeps your tests isolated from anything you are doing in dev mode, so your test data does not interfere with your dev mode data.

What would be happening here is that the test server would fail to start (due to the port being the same), and your tests run against the dev mode instance. This will cause problems as it will be using dev profile settings, and the continuous testing tracing (that figures out which tests affect which code) won't work.

Reply:
Continuous testing runs in a fully isolated class loader, so things like static variables etc from dev mode are isolated from the tests.

Reply:
> Continuous testing runs in a fully isolated class loader, so things like static variables etc from dev mode are isolated from the tests.

That answers the question somewhat.  Does this mean there are two Quarkus service instances running too?

Reply:
Yes, you can identify if you are in the secondary application via `io.quarkus.deployment.builditem.LaunchModeBuildItem#isAuxiliaryApplication` although for the most part things should be totally isolated so it does not matter. The only issue is socket bindings.

Note that even without the continuous testing support we would still want to run the tests on a different port to dev mode, so you can run the tests while you have dev mode running (this is why for HTTP tests are run on 8081 and dev mode on 8080).

Reply:
@stuartwdouglas I'm not seeing the behavior you're describing.  I set up my code to run on 8080 by default in dev mode and 8081 if in test mode.  When I enable continuous testing after a mvn quarkus:dev my rest-assured tests fail with a connection timeout.  I don't think the DevService is being re-run in an isolated class loader in this case.

Our am I really confused and you're just talking about doing continuous testing in one window and right-click IDE testing in another?

Reply:
@stuartwdouglas Nevermind, just bug (and superstition) on my part.